File: cpp/samples/update_thing_shadow.cpp
// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

// Example: Update a thing shadow

#include <gg/ipc/client.hpp>
#include <cstdlib>
#include <iostream>

int main() {
    auto &client = gg::ipc::Client::get();

    auto error = client.connect();
    if (error) {
        std::cerr << "Failed to establish IPC connection.\n";
        exit(-1);
    }

    std::string_view thing_name = std::getenv("AWS_IOT_THING_NAME");
    std::optional<std::string_view> shadow_name = "my-shadow";
    std::string_view payload = R"({"state":{"reported":{"temperature":25}}})";

    error = client.update_thing_shadow(
        thing_name, shadow_name, gg::Buffer { payload }
    );
    if (error) {
        std::cerr << "Failed to update shadow.\n";
        exit(-1);
    }

    std::cout << "Shadow updated successfully.\n";
}

File: python/samples/update_thing_shadow.py
# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
# SPDX-License-Identifier: Apache-2.0

# Example: Update a thing shadow

import os

import gg_sdk

sdk = gg_sdk.Sdk()
sdk.connect()

thing_name = os.environ["AWS_IOT_THING_NAME"]

sdk.update_thing_shadow(
    thing_name,
    b'{"state":{"reported":{"temperature":25}}}',
    shadow_name="my-shadow",
)

print("Shadow updated successfully.")
